program/src/matching.rs

```diff
@@ -75,17 +87,30 @@ impl InnerNode {
         let crit_bit = (search_key & crit_bit_mask) != 0;
         (self.children[crit_bit as usize], crit_bit)
     }
+
+    /// The lowest timestamp at which one of the contained LeafNodes expires.
+    #[inline(always)]
+    pub fn lowest_expiry(&self) -> u64 {
```

Comment:
nitpick: I was thinking if earliest vs latest are better than lowest vs highest, feel free to reject